DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DelegateProgressController.cs

```diff
@@ -91,5 +93,53 @@ DelegateProgressAccessRoute accessedVia
 
             return RedirectToAction("Index", "ViewDelegate", new { formData.DelegateId });
         }
+
+        [HttpGet]
+        [Route("EditCompletionDate")]
+        public IActionResult EditCompletionDate(int progressId, DelegateProgressAccessRoute accessedVia)
+        {
+            var centreId = User.GetCentreId();
+            var delegateCourseProgress =
+                courseService.GetDelegateCourseProgress(progressId, centreId);
+
+            var model = new EditCompletionDateViewModel(progressId, accessedVia, delegateCourseProgress!.DelegateCourseInfo);
+            return View(model);
+        }
+
+          [HttpPost]
+          [Route("EditCompletionDate")]
+          public IActionResult EditCompletionDate(
+              EditCompletionDateFormData formData,
+              int progressId,
+              DelegateProgressAccessRoute accessedVia)
+          {
+              if (!ModelState.IsValid)
+              {
+                  //todo not passing courseName back in so it disappears when we error

```

Comment:
Is there a reason this is a todo and not fixed for this action? We just need to add the hidden inputs to post the values back. (I know I need to fix this for mine too)

Reply:
fixed now

Reply:
Looks like the todo is still there?
Underlying cause is fixed.

Reply:
oops fixed now